packages/flutter_test/lib/src/finders.dart

```diff
@@ -511,6 +530,40 @@ class _WidgetIconFinder extends MatchFinder {
   }
 }
 
+class _WidgetWithIconFinder extends Finder {
```

Comment:
Can the logic in this class be simplified by reusing [find.descendant](https://docs.flutter.io/flutter/flutter_test/CommonFinders/descendant.html)?

Reply:
I don't think there's a nice way of reusing finders in other finders is there?

Reply:
`Finder widgetWithIcon(Type widgetType, IconData icon, ...) => find.ancestor(of: find.byIcon(icon), matching: find.byType(widgetType), ...);`

Reply:
Woah, awesome. Deleted the _WidgetWithTextFinder too